comms/src/outbound_message_service/message_pool_worker.rs

```diff
@@ -157,28 +189,30 @@ where
         msg: &mut OutboundMessage<MessageEnvelope>,
     ) -> Result<(), OutboundError>
     {
-        // Should attempt to connect and send, just mark as failed attempt
-        let mut peer = self.peer_manager.find_with_node_id(&msg.destination_node_id)?;
-        let net_address = peer.addresses.get_best_net_address()?;
-        self.peer_manager.mark_failed_connection_attempt(&net_address)?;
+        let peer = self.peer_manager.find_with_node_id(&msg.destination_node_id)?;
+        let peer_connection = self.connection_manager.establish_connection_to_peer(&peer)?;
+        peer_connection.send(msg.message_envelope.clone().into_frame_set())?;
```

Comment:
Can combine ln 194 and 196, by removing the `?;` and the`Ok(())`

Reply:
Done